evil-collection-buff-menu.el

```diff
@@ -27,10 +27,83 @@
 ;;; Bindings for `buff-menu'.
 
 ;;; Code:
+
+;; taken from emacs upstream repository with a minnor change
+;; this is all redundant code after emacs 26 
+(require 'tabulated-list) ;; required for evil-collection-Buffer-menu-unmark-all-buffers
+
+(defsubst evil-collection-tabulated-list-header-overlay-p (&optional pos)
+  "Return non-nil if there is a fake header.
+Optional arg POS is a buffer position where to look for a fake header;
+defaults to `point-min'."
+  (overlays-at (or pos (point-min))))
+
+(defun evil-collection-Buffer-menu-unmark-all ()
+  "Cancel all requested operations on buffers."
+  (interactive)
+  (evil-collection-Buffer-menu-unmark-all-buffers ?\r))
+
+(defun evil-collection-Buffer-menu-unmark-all-buffers (mark)
+  "Cancel a requested operation on all buffers.
+MARK is the character to flag the operation on the buffers.
+When called interactively prompt for MARK;  RET remove all marks."
+  (interactive "cRemove marks (RET means all):")
+  (save-excursion
+    (goto-char (point-min))
+    (when (evil-collection-tabulated-list-header-overlay-p)
+      (forward-line))
+    (while (not (eobp))
+      (let ((xmarks (list (aref (tabulated-list-get-entry) 0)
+			  (aref (tabulated-list-get-entry) 2))))
+	(when (or (char-equal mark ?\r)
+		  (member (char-to-string mark) xmarks))
+	  (Buffer-menu--unmark)))
+      (forward-line))))
+
 (defun evil-collection-buff-menu-setup ()
   "Set up `evil' bindings for `buff-menu'.."
-  ;; FIXME: Fill me in!
-  (evil-add-hjkl-bindings Buffer-menu-mode-map 'motion))
+
+  (evil-set-initial-state 'Buffer-menu-mode 'normal)
+  (evil-add-hjkl-bindings Buffer-menu-mode-map 'normal)
```

Comment:
This should be removed.

Reply:
the `(evil-add-hjkl-bindings Buffer-menu-mode-map 'normal)` ? It is that or add `hjkl` manually.

Reply:
By 'this', I mean the 'evil-add-hjkl-bindings' line.

Reply:
You shouldn't need to add hjkl manually in normal state.

Reply:
Evil keybinds aren't getting loaded in buff-mode, that applies specially to "k" and "v", those were binded by default in buff-mode.el. I tested it with only evil and evil-collection loaded.

Reply:
Can you show the exact snippet you used to test?

Reply:
The init I used was
```
(package-initialize)

(add-to-list 'package-archives
             '("melpa" . "https://melpa.org/packages/") t)

(unless (package-installed-p 'use-package)	;Auto install use-package.
  (progn (package-refresh-contents)
         (package-install 'use-package))
  )

(use-package evil-collection
  :load-path "~/.emacs.d/elisp/evil-collection/"
  :after evil
  :commands (evil-collection-init)
  :init
  (evil-collection-init)
  )

(use-package evil
  :ensure t
  :init
  (setq evil-want-integration nil)
  (evil-mode)
  )
```

Reply:
Lets keep it in for now then, I'll take a closer look at it later to see why it's not getting bound correctly without hjkl.